R/module-template.R

```diff
@@ -68,173 +68,173 @@ setMethod(
     filenameR <- file.path(nestedPath, paste0(name, ".R"))
 
     cat("
-# Everything in this file gets sourced during simInit, and all functions and objects
-#  are put into the simList. To use objects and functions, use sim$xxx.
-defineModule(sim, list(
-  name = \"", name, "\",
-  description = \"insert module description here\",
-  keywords = c(\"insert key words here\"),
-  authors = c(person(c(\"First\", \"Middle\"), \"Last\", email=\"email@example.com\", role=c(\"aut\", \"cre\"))),
-  childModules = character(),
-  version = numeric_version(\"0.0.0\"),
-  spatialExtent = raster::extent(rep(NA_real_, 4)),
-  timeframe = as.POSIXlt(c(NA, NA)),
-  timeunit = NA_character_, # e.g., \"year,\",
-  citation = list(\"citation.bib\"),
-  documentation = list(\"README.txt\", \"", name, ".Rmd\"),
-  reqdPkgs = list(),
-  parameters = rbind(
-    #defineParameter(\"paramName\", \"paramClass\", value, min, max, \"parameter description\")),
-    defineParameter(\".plotInitialTime\", \"numeric\", NA, NA, NA, \"This describes the simulation time at which the first plot event should occur\"),
-    defineParameter(\".plotInterval\", \"numeric\", NA, NA, NA, \"This describes the simulation time at which the first plot event should occur\"),
-    defineParameter(\".saveInitialTime\", \"numeric\", NA, NA, NA, \"This describes the simulation time at which the first save event should occur\"),
-    defineParameter(\".saveInterval\", \"numeric\", NA, NA, NA, \"This describes the simulation time at which the first save event should occur\")
-  ),
-  inputObjects = data.frame(
-    objectName = NA_character_,
-    objectClass = NA_character_,
-    sourceURL = \"\",
-    other = NA_character_,
-    stringsAsFactors = FALSE
-  ),
-  outputObjects = data.frame(
-    objectName = NA_character_,
-    objectClass = NA_character_,
-    other = NA_character_,
-    stringsAsFactors = FALSE
-  )
-))
-
-## event types
-#   - type `init` is required for initiliazation
-
-doEvent.", name, " = function(sim, eventTime, eventType, debug = FALSE) {
-  if (eventType == \"init\") {
-    ### check for more detailed object dependencies:
-    ### (use `checkObject` or similar)
-
-    # do stuff for this event
-    sim <- sim$", name, "Init(sim)
-
-    # schedule future event(s)
-    sim <- scheduleEvent(sim, params(sim)$", name, "$.plotInitialTime, \"", name, "\", \"plot\")
-    sim <- scheduleEvent(sim, params(sim)$", name, "$.saveInitialTime, \"", name, "\", \"save\")
-  } else if (eventType == \"plot\") {
+        # Everything in this file gets sourced during simInit, and all functions and objects
```

Comment:
all of the added indenting in this file may mess up the indenting of the file produced by `cat()`